lib/src/audit.dart

```diff
@@ -5,6 +5,45 @@ import 'dart:async';
 
 import 'from_handlers.dart';
 
+extension Audit<T> on Stream<T> {
+  /// Returns a Stream which only emits once per [duration], at the end of the
+  /// period.
+  ///
+  /// Always introduces a delay of at most [duration].
```

Comment:
It's really `at least` not `at most`? Other async/sync work can starve this for an arbitrary amount of time.

Reply:
I think it's tricky to describe. "Always introduces a delay of at most [duration] plus arbitrary delay for unrelated async work." might be the most accurate.

Some events won't be emitted. For the events that _are_ emitted, if it was the event that triggered the timer it will be delayed by `duration + arbitrary_async`. If it was an event that came in within that period it's delayed by `(duration - delta_since_event_triggering_timer) + arbitrary_async`...

Reply:
I updated the comment with a longer explanation and a timeline example. Does it make more sense this way?